mirrord/agent/src/main.rs

```diff
@@ -66,27 +69,42 @@ mod outgoing;
 mod runtime;
 mod sniffer;
 mod steal;
+mod tls_verifier;
 mod util;
 mod watched_task;
 
+/// Size of [`mpsc`] channels connecting [`TcpStealerApi`] and [`TcpSnifferApi`] with their
+/// background tasks.
 const CHANNEL_SIZE: usize = 1024;
 
-/// Keeps track of connected clients.
-#[derive(Debug)]
+/// Keeps track of next client id.
+/// Stores common data used when serving client connections.
+/// Can be cheaply cloned and passed to per-client background tasks.
+#[derive(Clone)]
 struct State {
-    clients: HashSet<ClientId>,
-    index_allocator: IndexAllocator<ClientId, 100>,
+    /// [`ClientId`] for the next client that connects to this agent.
+    next_client_id: Arc<AtomicU32>,
```

Comment:
Was this the last place we were using the `IndexAllocator`?

Reply:
Nope, agent's `FileManager`, `TcpConnectionSniffer` and task for outgoing UDP still use it